### ISSUE TITLE
[RFE] rpmbuild should present machine parseable failure reports

### ISSUE THREAD

**Issue by pemensik** [rpm-software-management/rpm#2769]
On talk about AI by Tomáš Tomeček I have realized rpm building process it quite terrible at providing machine processable results for failed builds. I think it should be reported in machine parseable format, JSON for example.

When the build has failed on BuildRequires: installation, it should even know exact list of packages missing to be working. But the list is just written into the output in unstructured way. I think optional status file should be provided, with mentioning in which section the failure happened and optional parameters, if it cat gather them.

I think those section failures should be reported separately:
- BuildRequires: missing (with list of packages)
- source files missing (with list of files)
- patch files missing (with list of files)
- patch apply failed (with list of patches)
- ``%prep`` fail in general
- ``%build`` fail
- ``%check`` fail
- ``%install`` fail
- unpackaged files found (with list of files)

Those information could be the presented by build service to the user, ideally with possibility to read only failing section output if I wish. If the tool has structured information, it should export it in structured way for machine processing. Such structured information could be used for statistics or faster fixing. In some cases it would allow even *automated* fixes of failing spec file.

**Comment by pemensik** [comment]
I think rpmbuild should return different return code at least for different sections, in which the failure occurred. But ideally also with lists of relevant object types it knows.

**Comment by pemensik** [comment]
At least it would be possible to say in which file to search for the error. Not something the end user has to find himself. Example: https://koji.fedoraproject.org/koji/taskinfo?taskID=105833248

But in general, I do not think the result should be pure boolean. More likely something like enum, with 0 used for success, positive numbers for issue type specification.

**Comment by ffesti** [comment]
I don't think we will get really machine parseable error reports as many errors are just spit out by random scripts. Also errors will almost always stop the build. So there will only be the first real error being reported.  But the error reporting can be better.
